Add double imperial tanks to defaults

MarkMarine opened the pull request:
When not using a dive computer that is air integrated, resetting the tank
size and nomenclature on each dive is tedious for imperial double tanks.
This change sets a standard for their nomenclature that works with the
XML export, and adds the typical tanks doubled by imperial divers to the
default list.

Signed-off-by: Mark Fox <mark.p.fox@gmail.com>

dirkhh commented:
So once you have such a tank / double in your divelog, it should be offered by Subsurface as one of the options in the dropdown. More precisely, once you add it and then restart Subsurface - I don't think this changes dynamically.
I'm worried about adding too many default tanks. I much rather have the really really common ones there and have the rest be picked up from a users dive file.

MarkMarine commented:
Maybe there is a bug in the mac version, but I can't add a new tank "type", I can only change the capacity of an existing tank "type". So in my log, I pick the HP100 "type" and change the size to 200 cuft, which to me isn't correct. Is this a bug, or is this the desired behavior?

MarkMarine commented:
I reduced the number of cylinders I added to the list, I still think it's valuable to have a few double imperial cylinders added to the default list, so users can easily see the naming convention that works with the XML export code.

sfuchs79 commented:
Ohoh, I feel a little bit guilty because I started with the 10l mono cylinders... ;-)
I would vote for adding some examples for every "family" of cylinders, so also for the double imperal.

BTW: IMHO one could even remove some of the metric doubles to have a shorter list. D7, D8.5, D12 and D18 are commonly used. D13, D15, D16 and D20 are maybe not so important.

Ghalr commented:
If I can give my opinion it would be interesting to investigate to know which cyl. size are most common, for example in France, Rec diving we have 12L / 15L 232bar only, most dive centers offer 15L in case unexperienced divers do not deal right with gas. For experienced/tech diving or divers who purchased their cyl. it should be possible to easily edit detailed cyl. specs. I mean, I assume that divers who make a trip in Europe will consider to travel by Car/Van than by plane, so they bring their cyl.

I believe 15L are more used than 18L.
10L, 12L, 15L, 18L should not be removed

MarkMarine commented:
I corrected the list of defaults per the suggestions here, and I added the most common double imperial cylinders. It still seems like there is a bug in adding a new cylinder to the log, I checked mac and linux and I couldn't edit the cylinder type or save an edited cylinder size, even after restarting.

Ghalr commented:
I found hard to understand how it works. Equipment tab right ??
- "Type" give me a lot of options, even with 12L etc
- "Size" give me 12L, 15L etc etc

I would be more interesting to have:
- A custom name for the tank we are editing
- Type / Name : Alu or Deco or Bottom for example
- Size : 10, 12, 15, 18L others   complete list
- Pressure : 200, 232, 300bar   complete list (only free)
- Add MOD ?
- Add END ?

jbygdell commented:
In order to create a custom cylinder all you have to do is type a name in the "Type" field, tabstep to size and type that in, tabstep to workingpressure and fill that in.
Once it is part of your logbook it will show up in the dropdown list.

MarkMarine commented:
That doesn't work on the most recent version on mac sierra. Check the video
I posted. Typing in the type box doesn't work.
On Sat, Mar 25, 2017 at 01:52 Joakim Bygdell <notifications@github.com>
wrote:

> In order to create a custom cylinder all you have to do is type a name in
> the "Type" field, tabstep to size and type that in, tabstep to
> workingpressure and fill that in.
> Once it is part of your logbook it will show up in the dropdown list.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Subsurface-divelog/subsurface/pull/272#issuecomment-289198870>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEXppCFbzDTs7Uqtb_nEopbZ4izUKgrMks5rpNXjgaJpZM4MnZ0i>
> .
>

jbygdell commented:
It does work on El Capitan.
My mac is to old to upgrade to Sierra so I can't verify the issue.

atdotde commented:
I can confirm the problem. Let's hope I find a bit of time to investigate this.

sfuchs79 commented:
Is the video above from planner mode or from main window/divelog mode?
I also had a look at this and can confirm for windows that you can't add a new cylinder while being in planner mode. In divelog mode everything works ok.

MarkMarine commented:
This is trying to edit an existing dive, but I've also seen the same
behavior in trying to plan a new dive.
On Sun, Mar 26, 2017 at 10:21 Stefan Fuchs <notifications@github.com> wrote:

> Is the video above from planner mode or from main window/divelog mode?
> I also had a look at this and can confirm for windows that you can't add a
> new cylinder while being in planner mode. In divelog mode everything works
> ok.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Subsurface-divelog/subsurface/pull/272#issuecomment-289300038>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEXppGgy13VT3HP947frUWnSidh_BkGsks5rpp6qgaJpZM4MnZ0i>
> .
>
